### ISSUE TITLE
Adding categorical crossentropy

### ISSUE THREAD

**Issue by cantlookback** [cantlookback/Perceptron#18]
Need to add categorical_crossentropy loss function.
